Library/ViewModels/PledgePaymentMethodsViewModel.swift

```diff
@@ -132,28 +127,37 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
   public let notifyDelegateApplePayButtonTapped: Signal<Void, Never>
   public let notifyDelegateCreditCardSelected: Signal<String, Never>
   public let notifyDelegateLoadPaymentMethodsError: Signal<String, Never>
-  public let reloadPaymentMethods: Signal<CardViewValues, Never>
+  public let reloadPaymentMethodsAndSelectCard:
+    Signal<([PledgeCreditCardViewData], GraphUserCreditCard.CreditCard?), Never>
 
   public let updateSelectedCreditCard: Signal<GraphUserCreditCard.CreditCard, Never>
 
   public var inputs: PledgePaymentMethodsViewModelInputs { return self }
   public var outputs: PledgePaymentMethodsViewModelOutputs { return self }
 }
 
-private func cardTypeAvailable(cards: [GraphUserCreditCard.CreditCard], availableCardTypes: [String])
-  -> [(card: GraphUserCreditCard.CreditCard, cardTypeIsAvailable: Bool)] {
-  var cardsWithIsAvailableCardType: [(GraphUserCreditCard.CreditCard, Bool)] = []
+private func pledgeCreditCardViewData(
+  with cards: [GraphUserCreditCard.CreditCard],
+  availableCardTypes: [String],
+  project: Project
+) -> [PledgeCreditCardViewData] {
+  return cards.compactMap { card -> PledgeCreditCardViewData? in
+    guard let cardBrand = card.type?.rawValue else { return nil }
 
-  cards.forEach { card in
-    guard let cardBrand = card.type?.rawValue else { return }
     let isAvailableCardType = availableCardTypes.contains(cardBrand)
-    cardsWithIsAvailableCardType.append((card, isAvailableCardType))
-  }
 
-  return cardsWithIsAvailableCardType
+    return (card, isAvailableCardType, project.location.displayableName)
+  }
 }
 
 private func showApplePayButton(for project: Project, applePayCapable: Bool) -> Bool {
   return applePayCapable &&
     AppEnvironment.current.config?.applePayCountries.contains(project.country.countryCode) ?? false
 }
+
+private func cards(
+  _ cards: [GraphUserCreditCard.CreditCard],
+  orderedBy backing: Backing?
+) -> [GraphUserCreditCard.CreditCard] {
+  return cards.sorted { card1, _ in card1.id == backing?.paymentSource?.id }
```

Comment:
Simple sort that places the card first if it's ID matches that of the payment source on the `Backing`.